src/renderer/app-update-warning/__tests__/app-update-warning.test.ts

```diff
@@ -0,0 +1,169 @@
+/**
```

Comment:
Monseigneur, all the unit tests here are technical, and really need to be written as behavioural instead :)

We can start considering the technical unit tests as rare exceptions, and behavioural ones as default.

Reply:
Can you give an example how behavioural version of such tests would be look like? Will it be some difference if we try to test how warning level would change over time?

Reply:
> Can you give an example how behavioural version of such tests would be look like?

I could if you really want, but are the current behaviours not a sufficient example already? I propose in a later comment that we look at this together :) So how about it, bud? :)

> Will it be some difference if we try to test how warning level would change over time?

Deeply sorry, I don't understand :(